experimental/LieAlgebras/src/LieAlgebra.jl

```diff
@@ -346,10 +390,28 @@ end
 
 Return `true` if `L` is abelian, i.e. $[L, L] = 0$.
 """
-function is_abelian(L::LieAlgebra)
+@attr Bool function is_abelian(L::LieAlgebra)
   return all(iszero, x * y for (x, y) in combinations(basis(L), 2))
```

Comment:
note part of this PR, but perhaps better as
```suggestion
  b = basis(L)
  n = length(b)
  return all(iszero, b[i] * b[j] for i in 1:n for j in i+1:n)
```

Just to say:
```
julia> b = [0,1,2,3];

julia> function f1(b)
         n = length(b)
         count(iszero, b[i] * b[j] for i in 1:n for j in i+1:n)
       end
f1 (generic function with 1 method)

julia> function f2(b)
         count(iszero, x * y for (x, y) in Oscar.combinations(b, 2))
       end
f2 (generic function with 1 method)

julia> @btime f1(b)
  51.735 ns (0 allocations: 0 bytes)
3

julia> @btime f1($b)
  9.217 ns (0 allocations: 0 bytes)
3

julia> @btime f2(b)
  693.878 ns (23 allocations: 1.70 KiB)
3

julia> @btime f2($b)
  658.019 ns (23 allocations: 1.70 KiB)
3
```